test/Test/Suggestion.hs

```diff
@@ -0,0 +1,72 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Test.Suggestion (tests) where
+
+import qualified Data.Text as T
+import Hedgehog (Gen, Group (..), MonadGen, Property, PropertyName, (===), (/==), checkParallel, forAll, property)
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+
+import Hintman.Suggestion.Core (Line, toLines)
+import Hintman.Suggestion.TrailingSpaces (suggest)
+
+data LineState = Plain | Trailing
+  deriving (Show)
+
+space :: MonadGen m => m Char
+space = pure ' '
+
+genSpaces :: Gen Text
+genSpaces = Gen.text (Range.linear 1 20) space
+
+genPlain :: Gen Text
+genPlain =
+  T.intercalate " " <$>
+  Gen.list (Range.linear 0 20) (
+      Gen.text (Range.linear 5 100) Gen.alpha)
+
+genLineStates :: Gen [LineState]
+genLineStates
```

Comment:
This generator is not used anywhere. Currently looks like you generate only plain lines or traling lines. Maybe you patch your data type a little bit:

```haskell
data Line
    = Plain Text
    = Trailing Text
```

So in that case you will store lines inside constructor